File: LeetCode/l860.cs
public class Solution {
    public bool LemonadeChange(int[] bills) {
        int l=bills.Length,c1=0,c2=0;
        for(int i=0;i<l;i++){
            if(bills[i]==5)c1++;
            else if(bills[i]==10){
                if(c1==0)return false;
                c1--;c2++;
            }else{
                if(c1>0&&c2>0){c1--;c2--;}
                else if(c1>3)c1-=3;
                else return false;
            }
        }
        return true; 
    }
}

File: LeetCode/l860.cpp
class Solution {
   public:
    bool lemonadeChange(vector<int>& bills) {
        int c5 = 0, c10 = 0;
        for (int i : bills) {
            if (i == 5)
                c5++;
            else if (i == 10) {
                if (c5 == 0)
                    return false;
                else {
                    c5--;
                    c10++;
                }
            } else {
                if (c5 && c10) {
                    c5--, c10--;
                } else if (c5 >= 3)
                    c5 -= 3;
                else
                    return false;
            }
        }
        return true;
    }
};